src/Form/ColumnsExtractor.php

```diff
@@ -56,4 +87,38 @@ public function getColumns($form)
 
         return $columns;
     }
+
+    /**
+     * Method adding custom fields labels to given array with column names.
+     * Method counts number of custom fields (result available via getCustomFieldsNumberOfColumns method).
+     *
+     * @param \Symfony\Component\Form\AbstractType $form    form, from which custom fields are extracted
+     * @param array                                $columns current array with column names (can be empty, then return array contains only custom fields)
+     *
+     * @return array array from input with custom fields attached
+     */
+    public function addCustomFieldsColumns($form, $columns = array())
```

Comment:
maybe add type hint for $form ?

Reply:
as discussed, we can leave it without type hinting for more flexibility